src/tests/ftest/io/nvme_pool_capacity.py

```diff
@@ -0,0 +1,255 @@
+        no_of_jobs = self.params.get("no_parallel_job", '/run/ior/*')
+        # Create a pool
+        pools = []
+        pool = {}
+
+        for val in range(0, num_pool):
+            pool[val] = TestPool(self.context,
+                                 dmg_command=self.get_dmg_command())
+            pool[val].get_params(self)
+            # Split the total SCM and NVME size for creating nultiple pools.
+            split_pool_scm_size = pool[val].scm_size.value / num_pool
+            split_pool_nvme_size = pool[val].nvme_size.value / num_pool
+            pool[val].scm_size.update(split_pool_scm_size)
+            pool[val].nvme_size.update(split_pool_nvme_size)
+            pool[val].create()
+            display_string = "pool{} space at the Beginning".format(val)
+            pool[val].display_pool_daos_space(display_string)
+            pools.append(pool[val])
+
+        self.log.info("Pools : %s", pools)
+
+        for test_loop in range(1):
+            self.log.info("--Test Repeat for loop %s---", test_loop)
+            # Create the IOR threads
+            threads = []
+            for thrd in range(no_of_jobs):
+                # Add a thread for these IOR arguments
+                threads.append(threading.Thread(target=self.ior_runner_thread,
+                                                kwargs={"test_case": test_case,
+                                                        "pool_array": pools,
+                                                        "num_jobs": no_of_jobs,
+                                                        "results":
+                                                        self.out_queue}))
+            # Launch the IOR threads
+            for thrd in threads:
+                self.log.info("Thread : %s", thrd)
+                thrd.start()
+                time.sleep(5)
+            # Wait to finish the threads
+            for thrd in threads:
+                thrd.join()
+
+            # Verify the queue and make sure no FAIL for any IOR run
+            while not self.out_queue.empty():
+                if self.out_queue.get() == "FAIL":
+                    self.fail("FAIL")
+        for val in range(0, num_pool):
+            display_string = "Pool{} space at the End".format(val)
+            pool[val].display_pool_daos_space(display_string)
+            pool[val].destroy()
+
+    def test_nvme_pool_capacity(self):
+        """Jira ID: DAOS-2085.
+
+        Test Description:
+            Purpose of this test is to verify whether DAOS stack
+            report NOSPC when accessing data beyond pool size.
+
+        Use case:
+        :avocado: tags=all,hw,large,nvme,pr
+        :avocado: tags=nvme_pool_capacity
+        """
+        # Use Case 1,2 and 3
+        self.test_run(1)
+        self.test_run(2, 2)
+
+        # Use Case create/delete pool/container
+        self.test_create_delete()
+        # self.test_create_delete(10, 50, 100)
+        # self.test_create_delete(10, 50, 100)
```

Comment:
(pylint-missing-final-newline) Final newline missing
(pylint-mixed-line-endings) Mixed line endings LF and CRLF